fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -832,6 +832,70 @@ ACTOR Future<Void> trackInitialShards(DataDistributionTracker* self, Reference<I
 	return Void();
 }
 
+ACTOR Future<Void> fetchTopKShardMetrics_impl(DataDistributionTracker* self, GetTopKMetricsRequest req) {
+	ASSERT(req.comparator);
+	try {
+		loop {
+			Future<Void> onChange;
+			std::vector<StorageMetrics> returnMetrics;
+
+			// TODO: shall we do random shuffle to make the selection uniform distributed over the shard space?
+			for (int i = 0; i < SERVER_KNOBS->DD_SHARD_COMPARE_LIMIT && i < req.keys.size(); ++i) {
```

Comment:
Not sure if this loop can be very long in time. If so, consider adding `yield` to avoid slow tasks.